Add macOS build note (#8115)
containerd/containerd: BUILDING.md

* Protoc 3.x compiler and headers (download at the [Google protobuf releases page](https://github.com/protocolbuffers/protobuf/releases))
* Btrfs headers and libraries for your distribution. Note that building the btrfs driver can be disabled via the build tag `no_btrfs`, removing this dependency.

> > *Note*: On macOS, building containerd is possible but running containerd natively is not yet supported on macOS.

Reviewer: ```suggestion
> *Note*: On macOS, you need a third party runtime to run containers on containerd
```

container runtime development. Make sure to follow the guidelines for versioning
in [RUNC.md](/docs/RUNC.md) for the best results.

> > *Note*: On macOS, runc is not supported.

Reviewer: ```suggestion
> *Note*: runc only supports Linux
```

Please refer to [RUNC.md](/docs/RUNC.md) for the currently supported version of `runc` that is used by containerd.

> > *Note*: On macOS, building containerd is possible but it will not contain a container runtime.

Reviewer 2: ```suggestion
> *Note*: On macOS, building containerd is possible but running containerd natively is not yet supported on macOS.
```

Reviewer 2: consider mentioning this issue up on the runc section as well .. cheers!

Reviewer 3: I agree with Mike's suggestion on how to word this, and that might just be due to the overloaded term container runtime lol. It's odd to state building containerd (which our About section on github is "An open and reliable container runtime") will not contain a container runtime.

Reviewer 3: @mxpv I'm curious on how you'd view wording this as well. It definitely "runs"

Reviewer 4: Yeah, I withheld comment worried I might appear as nitpicking, but now that @dcantah opened the floodgates :joy: ..we're kind of in a weird spot with macOS.. we could list a number of PRs that have continued to add viability to using containerd on macOS, especially for those who don't mind experimenting or picking up additional components outside of the containerd repo. Although this phrase ending with "not yet supported" might be a reasonable thing to say. It's hard to say how clear we should be here, but there is a difference between "it won't run" versus "we don't support it yet"

Reviewer 3: Yea it's hard to state support when there's no shim to point folks to. The daemon.. does daemon things and can pull images with the native snapshotter. Like you said there's a bit of a difference between "won't run", it definitely does that, but I'd be fine saying we don't really support it 🤷. As a fun side affair I was trying to get kata to work on darwin, so maybe if that ever gets finished we could change this?

Reviewer 5: Probably worth to distinguish containerd daemon here (which works just fine) and shim runtime for MacOS (which we don't have).

Reviewer 5: > You can build and run the containerd daemon on macOS, but there is no shim runtime currently offered by containerd maintainers that would let you run containers natively on this OS.

?

Please refer to [RUNC.md](/docs/RUNC.md) for the currently supported version of `runc` that is used by containerd.

> > *Note*: On macOS, the containerd daemon can be built and run natively. However, since runc is not supported natively, there will be no runtime engine to run containers. 

Reviewer: There are. Not runc though.

> > *Note*: On macOS, the containerd daemon can be built and run natively. However, since runc is not supported natively, there will be no runtime engine to run containers. 

Reviewer 2: suggest.. "However, as stated above, runc only supports linux."
